xla/service/gpu/runtime/command_buffer_cmd.cc

```diff
@@ -429,8 +430,13 @@ absl::Status TracedCommandBufferCmd::AddTracedCommandBuffer(
     const Thunk::ExecuteParams& execute_params,
     const RecordParams& record_params, se::CommandBuffer* command_buffer,
     absl::FunctionRef<absl::Status(se::Stream*)> trace) {
-  auto traced_cmd = record_params.state.GetOrCreate<TracedCommandBuffer>(
-      this, [&] { return std::make_unique<TracedCommandBuffer>(buffers()); });
+
+  const auto& debug_options = xla::GetDebugOptionsFromFlags();
```

Comment:
You should move `GetDebugOptionsFromFlags` into lambda because it can be very expensive and now you execute it on every graph re-capture